R/opendp/tests/testthat/test-measurements.R

```diff
@@ -24,54 +24,54 @@ test_that("make_laplace", {
     (space |> then_laplace(1.))(arg = c(0L, 1L))
 })
 
-test_that("make_base_geometric", {
+test_that("make_geometric", {
     space <- c(atom_domain(.T = "i32"), absolute_distance(.T = "i32"))
-    meas <- space |> then_base_geometric(1., bounds = c(0L, 2L))
+    meas <- space |> then_geometric(1., bounds = c(0L, 2L))
     expect_equal(typeof(meas(arg = 0L)), "integer")
     expect_equal(meas(d_in = 1L), 1.0)
     # print(jsonlite::toJSON(to_ast(meas), pretty = TRUE))
 })
 
-test_that("make_base_laplace", {
+test_that("make_laplace", {
     space <- c(atom_domain(.T = "f64"), absolute_distance(.T = "f64"))
-    meas <- space |> then_base_laplace(1., k = -40L)
+    meas <- space |> then_laplace(1., k = -40L)
     expect_equal(typeof(meas(arg = 0)), "double")
     expect_equal(meas(d_in = 1), 1.0)
 })
 
-test_that("make_base_discrete_laplace", {
+test_that("make_discrete_laplace", {
     space <- c(atom_domain(.T = "i32"), absolute_distance(.T = "i32"))
-    meas <- space |> then_base_discrete_laplace(1.)
+    meas <- space |> then_laplace(1.)
     expect_equal(typeof(meas(arg = 0L)), "integer")
     expect_equal(meas(d_in = 1L), 1.0)
 })
 
-test_that("test_base_gaussian_curve", {
+test_that("test_gaussian_curve", {
     input_space <- c(atom_domain(.T = f64), absolute_distance(.T = f64))
-    meas <- make_zCDP_to_approxDP(input_space |> then_base_gaussian(4.))
+    meas <- make_zCDP_to_approxDP(input_space |> then_gaussian(4.))
     curve <- meas(d_in = 1.)
     expect_equal(curve(delta = 0.), Inf)
     expect_equal(curve(delta = 1e-3), 0.6880024554878086)
     expect_equal(curve(delta = 1.), 0.)
 
-    curve <- make_zCDP_to_approxDP(input_space |> then_base_gaussian(4.))(d_in = 0.0)
+    curve <- make_zCDP_to_approxDP(input_space |> then_gaussian(4.))(d_in = 0.0)
     expect_equal(curve(delta = 0.0), 0.0)
     expect_error(curve(delta = -0.0))
 
-    curve <- make_zCDP_to_approxDP(input_space |> then_base_gaussian(0.))(d_in = 1.0)
+    curve <- make_zCDP_to_approxDP(input_space |> then_gaussian(0.))(d_in = 1.0)
     expect_equal(curve(delta = 0.0), Inf)
     expect_equal(curve(delta = 0.1), Inf)
 
-    curve <- make_zCDP_to_approxDP(input_space |> then_base_gaussian(0.))(d_in = 0.0)
+    curve <- make_zCDP_to_approxDP(input_space |> then_gaussian(0.))(d_in = 0.0)
     expect_equal(curve(delta = 0.0), 0.0)
     expect_equal(curve(delta = 0.1), 0.0)
 })
 
-test_that("test_base_gaussian_search", {
+test_that("test_gaussian_search", {
```

Comment:
Ideally, the substantive tests are all handled in Rust, and the tests from bindings languages are stripped down to be as thin as possible, so that they just test the bindings. This should make it possible to automate test generation as well. Codegen for tests is a low priority, but this policy toward testing will make it less of a lift to automate in the future. 

Going by this approach, then yes, new tests for functionality in Rust should match each other in R and Python, be as thin as possible in R and Python, and be concentrated more in Rust.